exportoptionsgenerator/exportoptionsgenerator.go

```diff
@@ -53,14 +52,60 @@ func New(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configuration
 }
 
 // GenerateApplicationExportOptions generates exportOptions for an application export.
-func (g ExportOptionsGenerator) GenerateApplicationExportOptions(exportMethod exportoptions.Method, containerEnvironment string, teamID string, uploadBitcode bool, compileBitcode bool, xcodeManaged bool,
-	xcodeMajorVersion int64) (exportoptions.ExportOptions, error) {
+func (g ExportOptionsGenerator) GenerateApplicationExportOptions(
+	exportMethod exportoptions.Method,
+	containerEnvironment string,
+	teamID string,
+	uploadBitcode bool,
+	compileBitcode bool,
+	archivedWithXcodeManagedProfiles bool,
+	codeSigningStyle SigningStyle,
+	xcodeMajorVersion int64,
+) (exportoptions.ExportOptions, error) {
+	mainTargetBundleID, entitlementsByBundleID, err := g.applicationTargetsAndEntitlements(exportMethod)
```

Comment:
Code from the previous `g.generateExportOptions` moved here the first part of the code is moved into the new `applicationTargetsAndEntitlements`.